File: src/main/java/com/optica/services/ApplyService.java
package com.optica.services;

import java.util.List;

import org.springframework.data.domain.Pageable;

import com.optica.domain.Apply;

public interface ApplyService {

	List<Apply> getAppliesByCurriculum(Long id,Pageable pageable);
	
	List<Apply> getAppliesByJobOffer(Long id,Pageable pageable);
	
	void applyForJobOffer(Apply apply);
	
	void deleteApply(Long id);
	
}

File: src/main/kotlin/com/optica/services/ApplyService.kt
package com.optica.services

import org.springframework.data.domain.Pageable

import com.optica.domain.Apply

interface ApplyService {

    fun getAppliesByCurriculum(id: Long?, pageable: Pageable): List<Apply>

    fun getAppliesByJobOffer(id: Long?, pageable: Pageable): List<Apply>

    fun applyForJobOffer(apply: Apply)

    fun deleteApply(id: Long?)

}
